qiita_db/data.py

```diff
@@ -1257,6 +1257,60 @@ def create(cls, processed_params_table, processed_params_id, filepaths,
         pd.add_filepaths(filepaths, conn_handler)
         return cls(pd_id)
 
+    @classmethod
+    def delete(cls, processed_data_id):
+        """Removes the processed data with id processed_data_id
+
+        Parameters
+        ----------
+        processed_data_id : int
+            The processed data id
+
+        Raises
+        ------
+        QiitaDBUnknownIDError
+            If the processed data id doesn't exist
+        QiitaDBStatusError
+            If the processed data status is not sandbox
+        """
+        if cls(processed_data_id).status != 'sandbox':
+            raise QiitaDBStatusError(
+                "Illegal operation on non sandbox processed data")
+
+        conn_handler = SQLConnectionHandler()
+
+        analysis_ids = [str(_id[0]) for _id in conn_handler.execute_fetchall(
+            "SELECT DISTINCT analysis_id FROM qiita.analysis_sample WHERE "
+            "processed_data_id = {0} ORDER BY "
+            "analysis_id".format(processed_data_id))]
+
+        if analysis_ids:
+            raise QiitaDBError(
+                "Processed data %d is linked to (meta)analyses: %s" %
```

Comment:
And users should not be able to put it back to sandbox if it was public and
there was analysis using it.

2015-04-03 9:31 GMT-07:00 Antonio Gonzalez <notifications@github.com>:

> In qiita_db/data.py
> <https://github.com/biocore/qiita/pull/1034#discussion_r27738714>:
>
> > +            If the processed data status is not sandbox
> > +        """
> > +        if cls(processed_data_id).status != 'sandbox':
> > +            raise QiitaDBStatusError(
> > +                "Illegal operation on non sandbox processed data")
> > +
> > +        conn_handler = SQLConnectionHandler()
> > +
> > +        analysis_ids = [str(_id[0]) for _id in conn_handler.execute_fetchall(
> > +            "SELECT DISTINCT analysis_id FROM qiita.analysis_sample WHERE "
> > +            "processed_data_id = {0} ORDER BY "
> > +            "analysis_id".format(processed_data_id))]
> > +
> > +        if analysis_ids:
> > +            raise QiitaDBError(
> > +                "Processed data %d is linked to (meta)analyses: %s" %
>
> I think that's fine. The only way this could happen is if the owner made
> it public, then move it to sandbox, and is trying to delete it. Which will
> be a full other beast ...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/qiita/pull/1034/files#r27738714>.
>



-- 
Jose Navas